src/net_processing.cpp

```diff
@@ -1373,7 +1375,11 @@ static void RelayAddress(const CAddress& addr, bool fReachable, const CConnman&
     // Use deterministic randomness to send to the same nodes for 24 hours
     // at a time so the m_addr_knowns of the chosen nodes prevent repeats
     uint64_t hashAddr = addr.GetHash();
-    const CSipHasher hasher = connman.GetDeterministicRandomizer(RANDOMIZER_ID_ADDRESS_RELAY).Write(hashAddr << 32).Write((GetTime() + hashAddr) / (24 * 60 * 60));
+
+    std::chrono::seconds current_time = GetTime<std::chrono::seconds>();
+    // Adding address hash makes exact rotation time different per address, while preserving periodicity.
+    uint64_t time_addr = (current_time + std::chrono::seconds(hashAddr)) / ROTATE_ADDR_RELAY_DEST_INTERVAL;
```

Comment:
```suggestion
    uint64_t time_addr = (current_time + std::chrono::seconds{static_cast<int64_t>(hashAddr)}) / ROTATE_ADDR_RELAY_DEST_INTERVAL;
```

Maybe try this to avoid https://travis-ci.org/github/bitcoin/bitcoin/jobs/675780723#L4756